api/edge_api/identities/export.py

```diff
@@ -0,0 +1,205 @@
+
+EXPORT_EDGE_IDENTITY_PAGINATION_LIMIT = 20000
+
+logger = logging.getLogger()
+
+
+def export_edge_identity_and_overrides(  # noqa: C901
+    environment_api_key: str,
+) -> Tuple[list, list, list]:
+    kwargs = {
+        "environment_api_key": environment_api_key,
+        "limit": EXPORT_EDGE_IDENTITY_PAGINATION_LIMIT,
+    }
+    identity_export = []
+    traits_export = []
+    identity_override_export = []
+
+    feature_id_to_uuid: Dict[int, str] = {}
+
+    mv_feature_option_id_to_uuid: Dict[int, str] = {}
+    while True:
+        response = EdgeIdentity.dynamo_wrapper.get_all_items(**kwargs)
+        for item in response["Items"]:
+            identifier = item["identifier"]
+            # export identity
+            identity_export.append(
+                export_edge_identity(
+                    identifier, environment_api_key, item["created_date"]
+                )
+            )
+            # export traits
+            for trait in item["identity_traits"]:
+                traits_export.append(
+                    export_edge_trait(trait, identifier, environment_api_key)
+                )
+            for override in item["identity_features"]:
+                featurestate_uuid = override["featurestate_uuid"]
+                feature_id = override["feature"]["id"]
+                if feature_id not in feature_id_to_uuid:
+                    try:
+                        feature_id_to_uuid[feature_id] = _get_feature_uuid(feature_id)
+                    except Feature.DoesNotExist:
+                        logging.warning("Feature with id %s does not exist", feature_id)
+                        continue
+
+                feature_uuid = feature_id_to_uuid[feature_id]
+
+                # export feature state
+                identity_override_export.append(
+                    export_edge_feature_state(
+                        identifier,
+                        environment_api_key,
+                        featurestate_uuid,
+                        feature_uuid,
+                        override["enabled"],
+                    )
+                )
+                featurestate_value = override["feature_state_value"]
+                if featurestate_value is not None:
+                    # export feature state value
+                    identity_override_export.append(
+                        export_featurestate_value(featurestate_value, featurestate_uuid)
+                    )
+                if mvfsv := override["multivariate_feature_state_values"]:
+                    # only one mvfs override can exists
+                    mv_feature_option_id = mvfsv[0]["multivariate_feature_option"]["id"]
+                    if mv_feature_option_id not in mv_feature_option_id_to_uuid:
+                        try:
+                            mv_feature_option_id_to_uuid[mv_feature_option_id] = (
+                                _get_mv_feature_option_uuid(mv_feature_option_id)
```

Comment:
As per the above, maybe we should grab the options up front and use it here, instead of always calling `.get` in the function.

Reply:
We are calling get only if the `id` is not in our cache (mv_feature_option_id_to_uuid), so the only difference is that we are populating the cache lazily

Reply:
But if there are 600 features, each with at least 1 override, we are making 600 queries... vs just one to grab all the features up front?

Reply:
Yep, Done